src/libraries/System.Security.Cryptography.Encoding/tests/Cbor.Tests/CborWriterTests.Integer.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+using System;
+using Test.Cryptography;
+using Xunit;
+
+namespace System.Security.Cryptography.Encoding.Tests.Cbor
+{
+    public partial class CborWriterTests
+    {
+        // Data points taken from https://tools.ietf.org/html/rfc7049#appendix-A
+
+        [Theory]
+        [InlineData(0, "00")]
+        [InlineData(1, "01")]
+        [InlineData(10, "0a")]
+        [InlineData(23, "17")]
+        [InlineData(24, "1818")]
+        [InlineData(25, "1819")]
+        [InlineData(100, "1864")]
+        [InlineData(1000, "1903e8")]
+        [InlineData(1000000, "1a000f4240")]
+        [InlineData(1000000000000, "1b000000e8d4a51000")]
+        [InlineData(-1, "20")]
+        [InlineData(-10, "29")]
+        [InlineData(-100, "3863")]
+        [InlineData(-1000, "3903e7")]
+        public static void Int64WriterTests(long input, string hexExpectedEncoding)
+        {
+            byte[] expectedEncoding = hexExpectedEncoding.HexToByteArray();
+            using var writer = new CborWriter();
+            writer.Write(input);
+            Assert.Equal(expectedEncoding, writer.ToArray());
```

Comment:
Generally I've found the binary protocols nicer to assert the hex strings are equal, because I find it easier to interpret (and relate to a test case)

```
Assert.Equals Failure
    Expected: "...00E8D4A51000"
      Actual: "...00E8D5A51000"
                       ^
```

Than

```
Assert.Equals Failure
    Expected: { ..., 0, 232, 212, 165, 16, 0 }
      Actual: { ..., 0, 232, 213, 165, 16, 0 }
```


That's assuming that xUnit has even fixed the array-mismatch string (it used to always just show the first N elements, not where the mismatch was).